astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -2298,7 +2298,12 @@ def infer_call_result(self, caller, context=None):
         try:
             metaclass = self.metaclass(context=context)
             if metaclass is not None:
-                dunder_call = next(metaclass.igetattr("__call__", context))
+                # Only get __call__ if it's defined locally for the metaclass.
+                # Otherwise we will find ObjectModel.__call__ which will
+                # return an instance of the metaclass. Instantiating the class is
+                # handled later.
+                if "__call__" in metaclass.locals:
+                    dunder_call = next(metaclass.igetattr("__call__", context))
```

Comment:
Should there be a default value for next?

Reply:
Caught by `StopIteration` on L2307.

Reply:
Thanks, didn't see it!